state/state.go

```diff
@@ -0,0 +1,77 @@
+package state
+
+import (
+	"log"
+	"sync"
+
+	"github.com/NordSecurity/nordvpn-linux/internal"
+)
+
+type Event int
+
+const (
+	VPNConnected Event = iota
+	VPNDisconnected
+)
+
+type subscriber struct {
+	stateChan chan<- Event
+	stopChan  <-chan struct{}
+}
+
+func newSubscriber(stateChan chan<- Event, stopChan <-chan struct{}) subscriber {
+	return subscriber{
+		stateChan: stateChan,
+		stopChan:  stopChan,
+	}
+}
+
+type subscriberSet map[subscriber]bool
+
+type StatePublisher struct {
+	mu          sync.Mutex
+	subscribers subscriberSet
+}
+
+func NewState() StatePublisher {
+	return StatePublisher{
+		subscribers: make(subscriberSet),
+	}
+}
+
+func (s *StatePublisher) NotifyVPNInternalStateChange(e Event) error {
+	s.mu.Lock()
+	defer s.mu.Unlock()
+
+	log.Println(internal.InfoPrefix+" Received VPN state change notification: ", e)
+
+	for sub := range s.subscribers {
+		select {
+		case <-sub.stopChan:
+			delete(s.subscribers, sub)
+		case sub.stateChan <- e:
+		}
+	}
+
+	return nil
+}
+
+func (s *StatePublisher) AddSubscriber() (<-chan Event, chan<- struct{}) {
```

Comment:
This can be a regular `subject.Subject` which has a subscriber that sends events via gRPC on notify